pulsar-client/src/main/java/org/apache/pulsar/client/impl/conf/ClientConfigurationData.java

```diff
@@ -133,5 +142,24 @@ public ClientConfigurationData clone() {
         }
     }
 
+    public InetSocketAddress getSocks5ProxyAddress() {
+        if (Objects.nonNull(socks5ProxyAddress)) {
+            return socks5ProxyAddress;
+        }
+        String proxyAddress = System.getProperty("socks5Proxy.address");
```

Comment:
Does it need to config client by the system property?

Reply:
Yes, this is not a common feature, @codelipenghui suggest to support from System property. But keep the clientBuilder together. We maybe remove the client builder api somedays.

Reply:
ok